### ISSUE TITLE
Allow inserting of custom SQF functions

### ISSUE THREAD

**Issue by dedmen** [intercept/intercept#109]
It's possible to call the functions to add functions/operators/nulars to the engine and then call the functions from SQF directly. This will allow for tighter integration of C++ and SQF. 
One can already signal from SQF to his C++ code by using callExtension.. But this causes some overhead. Just adding a nular or unary function to do that signal would circumvent all the overhead of callExtension and parsing the command.
Basically calling directly to custom C++ code without going through callExtension.
Would also make possible to implement some composite functions that just execute multiple SQF functions but in C++ context. So being faster than making these operations in SQF

**Comment by marceldev89** [comment]
I might be missing something but I think, with intercept, it would make more sense to go 100% intercept and fall back to SQF when needed with https://github.com/intercept/intercept/wiki/Inline-SQF. If you want to do SQF with some C++ it's probably better to just go with a "simple" extension instead.

**Comment by dedmen** [comment]
Most codebases won't be completly convertable to intercept. Many projects also have API functions in their SQF code that are called by the Mods users. Even if everything else is done in Intercept.. These API functions would benefit from a fast way to call into Intercept without going through callExtension.
Of cause one could just not implement API functions but then anyone who wanted to interact with the core of the Mod would have to also use a Intercept project.
Or for projects that will only partially use Intercept for example ace. you could just
if (interceptAvailable) then { call intercept function } else {do sqf stuff}
to improve performance if intercept is available. They could already do it by passing parameters through callExtension and then again parse the output of that. But then you have all the overhead of parsing that stuff.

**Comment by overfl0** [comment]
Isn't this a #10 duplicate?

**Comment by dedmen** [comment]
Correct. @overfl0

**Comment by dedmen** [comment]
Closing in favor pf #10
